templates/flutter/lib/src/realtime_mixin.dart.twig

```diff
@@ -76,6 +76,23 @@ mixin RealtimeMixin {
             }
             break;
         }
+      }, onDone: () {
+        for (var list in _channels.values) {
+          for (var stream in list) {
+            stream.close();
+          }
+        }
+        _closeConnection();
```

Comment:
I think we should try to reconnect if there are channels or something similar to how we do for [web](https://github.com/appwrite/sdk-for-web/blob/master/src/client.ts#L257)

My server times out realtime connections after 2 minutes and using web, it reconnects automatically. Using the flutter SDK, it doesn't try to reconnect.

Reply:
yes i'm struggling with that, can you try ?

Reply:
Try what? I was suggesting to reconnect in this callback if there were still channels instead of just closing.

Reply:
It might be counterproductive to try to reconnect in the callback if the connection was terminated when there was no internet connection?

Reply:
I think there may be other reasons why a connection might close, though.

Reply:
I guess we can revise this later.